todo_app/app.py

```diff
@@ -1,57 +1,106 @@
-from flask import Flask, request, render_template
+from flask import Flask, request, render_template, redirect
 from todo_app.flask_config import Config
 from todo_app.data.session_items import get_items, add_item, get_item, save_item, delete_item
 from operator import itemgetter
+import requests
+import json
 
 app = Flask(__name__)
 app.config.from_object(Config)
+trello_authorisation = {'key': Config.API_KEY,'token': Config.TOKEN}
 
+def get_status_name(card_id):
+    status_name = requests.get('https://api.trello.com/1/cards/' + card_id +'/list', params = trello_authorisation)
+    status_name_json = status_name.json()
+    return status_name_json["name"]
+
+def get_next_list(list_id):
+    next_list_response = requests.get('https://api.trello.com/1/boards/' + Config.TRELLO_BOARD + '/lists', params = trello_authorisation)
+    next_list_response_json = next_list_response.json()
+    use_next_list_entry = False
+    for mylist in next_list_response_json:
+        if use_next_list_entry == True:
+            use_next_list_entry = False
+            next_list_name = mylist["name"]
+            next_list_id = mylist["id"]
+        
+        if mylist["id"] == list_id:
+            use_next_list_entry = True
+    
+    ##the card is already on the final list e.g completed
+    if use_next_list_entry == True:
+        next_list_name = "Completed"
+        next_list_id = list_id
+    
+    next_list_dict = {'name':next_list_name, 'id': next_list_id}
```

Comment:
I would write this function more like this:

```
def get_next_list(list_id):
    lists_response = requests.get('https://api.trello.com/1/boards/' + Config.TRELLO_BOARD + '/lists', params = trello_authorisation)
    all_lists = lists_response.json()

    for index, mylist in enumerate(all_lists):
        if mylist["id"] == list_id and index < len(all_lists - 1):
            next_list = all_lists[index + 1]
            return {'name': next_list['name'], 'id': next_list['id']}
    
    return None
```

It is somewhat dishonest to return "Completed" and the current list id as the result. You can instead return None and then the thing trying to display next list can decide what to do with that. In this case, index.html shouldn't really render a button.

The API response is simply for lists not "next lists"

I've simplified the logic a bit.
Also small point with your code `if boolean_variable == True` can just be written as `if boolean_variable`